compiler/fir/cones/src/org/jetbrains/kotlin/fir/types/CompilerConeAttributes.kt

```diff
@@ -50,20 +50,32 @@ object CompilerConeAttributes {
     object ExtensionFunctionType : ConeAttribute<ExtensionFunctionType>() {
         val ANNOTATION_CLASS_ID = ClassId(FqName("kotlin"), Name.identifier("ExtensionFunctionType"))
 
-        override fun union(other: ExtensionFunctionType?): ExtensionFunctionType? = other
-        override fun intersect(other: ExtensionFunctionType?): ExtensionFunctionType? = this
+        override fun union(other: ExtensionFunctionType?): ExtensionFunctionType? = this
+        override fun intersect(other: ExtensionFunctionType?): ExtensionFunctionType? = other
```

Comment:
So, the underlying logic is, for the same key, we will retrieve `ConeAttribute` from both side, and only non-null result (of operation) will be recorded.

It is obvious that `this` is non-null. For `union`, returning non-null `this` will act like `or`. For `intersect`, returning the other one, which could be `null`, will act like `and`.

So, with previous implementation (that returns `other` for `union`, which could be `null`), I saw that `union` always returns `EMPTY` attributes!

Reply:
I strongly disagree with those changes. `union` and `intersect` define how attributes should be passed throw common super type calculation (`intersect` for intersection types and `union` for union types (which will be supported somewhere in future)). And for this attributes we have strong rules:
- `Exact` and `NoInfer` stays only on upper type and don't flow throw inference
- If on component of intersection type is `ExtensionFuncitonType` then the whole type also is

Reply:
Thanks for the detailed explanation. I somehow still/sometimes misunderstood `union` and `intersect` types. :\ If these attributes are used for such super type calculation, yeah, the original expression for `union`/`intersect` make sense. I just need to use `intersect` where I used `union` previously.